packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -109,11 +116,27 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
     private async readFromUserHome(): Promise<WorkspaceData | undefined> {
         const file = this.getUserStoragePath();
         if (await fs.pathExists(file)) {
-            const config = await fs.readJson(file);
+            const rawContent = await fs.readFile(file, 'utf-8');
+            const content = rawContent.trim();
+            if (!content) {
+                return undefined;
+            }
+
+            let config;
+            try {
+                config = JSON.parse(content);
+            } catch (error) {
+                error.message = `${file}:\n${error.message}`;
```

Comment:
I wonder if instead, in this case, we might just mention that the recentworkspace.json file has an issue and will be ignored? e.g.: "Error parsing /home/user/.theia/recentworkspace.json. File will be ignored". The user can look at the log for details or delete the file.